sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3799,23 +3801,26 @@ object functions {
     ArrayExcept(col1.expr, col2.expr)
   }
 
+  // counter to ensure lambda variable names are unique
+  private val lambdaVarNameCounter = new AtomicInteger(0)
```

Comment:
How about defining a companion object for UnresolvedNamedLambdaVariable like this?
```

object UnresolvedNamedLambdaVariable {

  // counter to ensure lambda variable names are unique
  private val lambdaVarNameCounter = new AtomicInteger(0)

  def apply(args: Seq[String]): UnresolvedNamedLambdaVariable = {
    // add a counter number in the suffix
  }
}
```